src/utils/tracked-controls.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Return enumerated gamepads matching id prefix.
+ *
+ * @param {object} idPrefix - prefix to match in gamepad id, if any.
+ */
+module.exports.getGamepadsByPrefix = function (idPrefix) {
+  var rtn = [];
+  var gamepads = navigator.getGamepads && navigator.getGamepads();
+  if (gamepads) {
+    for (var i = 0; i < gamepads.length; ++i) {
+      var gamepad = gamepads[i];
+      if (gamepad) {
+        if (!idPrefix || idPrefix === '' || gamepad.id.indexOf(idPrefix) === 0) {
```

Comment:
Could you do at the beginning of the function?
````javascript
if (!idPrefix) { return gamepads; }
```` 
and this `if` statement would just be:
````javascript
if (gamepad.id.indexOf(idPrefix) !== 0) { break; }
gamepadList.push(gamepad);
````